server/src/test/java/org/opensearch/action/admin/cluster/stats/ClusterStatsRequestTests.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.stats;
+
+import org.opensearch.Version;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.util.Set;
+
+public class ClusterStatsRequestTests extends OpenSearchTestCase {
```

Comment:
We've good tests in `AbstractWireSerializingTestCase` and `AbstractXContentTestCase`.
Can you check if the tests are useful, and we can reuse some of the predefined cases as they already cover a wide variety of Req/Resp ser/deser

Reply:
Added this test case to specifically test the ser/de od ClusterStateRequest object for versions greater than 3.0 and below 3.0 specifically. I checked the AbstractWireSerializingTestCase, these test case require equals method implementation in the class being tested. This class doesn't have equals method implementation and the class variable types also don't have an equals method implementation. Additionally the test cases in AbstractWireSerializingTestCase only test for current version where as for this change we want to test for versions both below and above 3.0.

Reply:
this test has to be updated when merging in 2.x.

Reply:
Acknowledge.